modules/components/src/AdvancedSqonBuilder/filterComponents/BooleanFilter.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Component from 'react-component-component';
+
+import {
+  getOperationAtPath,
+} from '../utils';
+import BooleanAgg from '../../Aggs/BooleanAgg';
+import { FilterContainer } from './common';
+import './FilterContainerStyle.css';
+
+const getFieldDisplayName = (fieldDisplayNameMap, initialFieldSqon) => {
+  return fieldDisplayNameMap[initialFieldSqon.content.field] || initialFieldSqon.content.field;
+};
+
+const AggsWrapper = ({ children }) => (
+  <div className="aggregation-card">{children}</div>
+);
+
+export const BooleanFilterUI = props => {
+  const {
+    onSubmit = sqon => {},
+    onCancel = () => {},
+    ContainerComponent = FilterContainer,
+    sqonPath = [],
+    initialSqon = null,
+    field = '',
+    fieldDisplayNameMap = {},
+  } = props;
+
+  const initialState = {
+    selectedValue: undefined,
+    localSqon: initialSqon
+  };
+
+  const initialFieldSqon = getOperationAtPath(sqonPath)(initialSqon) || {
+    op: 'and',
+    content: { value: initialState.selectedValue, field },
+  };
+
+  const onSqonSubmit = s => () => onSubmit(s.state.localSqon);
+
+  const onSelectionChange = s => ({ value, generateNextSQON }) => {
```

Comment:
I'm not sure exactly what you mean, could you provide a corrected version of `onSelectionChange` and explain under which circumstances it would fail?
Honestly, my understanding of that code is not rock solid, yet.

Reply:
So, the original `BooleanAgg` component was not built to work with non-nested sqons. This component however needs to be able to work with nested sqons that may look like:
```
{
  "op": "and",
  "content": [
    {
      "op": "and",
      "content": [
        ...someMoreQueries,
        {
          "op": "in",
          "content": {
            "field": "is_proband",
            "value": ["false"]
          }
        }
      ]
    }
  ]
}
```
by calling `generateNextSQON(s.state.localSqon)`, we are getting back a new sqon with the `is_proband in ...` only injected at the top level `and`, so it won't quite work for nested queries. 

Looking deeper into the code here, I think we don't quite need to use `generateNextSQON` in this particular component. It's simpler to generate a new `in` op from `value.key_as_string`, then inject it into `s.state.localSqon` at the `sqonPath` specified in props.
(`setSqonAtPath` from `utils.js` can be used for the the injection. It's non-mutative so you get back a new instance altogether).

This is complex, but the discussion around what level of support for nested sqons was a point of back-and-forth (and still may change in future) so I'm trying to keep it as flexible as possible.

Here's a blind attempt :)
```
const onSelectionChange = s => ({ value }) => {
    const newOp = {
        op: "in",
        content: {
            field, //this should be available from props
            value: [ value.key_as_string ]
        }
    }
    s.setState({
      localSqon: setSqonAtPath(sqonPath, newOp)(s.state.localSqon),
    });
  };
```

hope this helps :)

Reply:
I though that the role of `generateNextSQON` was to find the to inject or replace the local sqon. My bad, I'll change that.

Reply:
you're not wrong @jberube :) just that the original BooleanAgg component was not designed to handle nested sqons